manifests/plugin/host_reports.pp

```diff
@@ -0,0 +1,48 @@
+# = Foreman Proxy Host Reports plugin
+#
+# This class installs the Host Reports plugin
+#
+# === Parameters:
+#
+# $proxy_name::               Proxy hostname to appear in reports JSON
+#
+# $spool_dir::                Spool directory with processed reports
+#
+# === Advanced parameters:
+#
+# $enabled::                  enables/disables the host_reports plugin
+#
+# $listen_on::                proxy feature listens on http, https, or both
+#
+# $version::                  plugin package version, it's passed to ensure parameter of package resource
+#                             can be set to specific version number, 'latest', 'present' etc.
+#
+class foreman_proxy::plugin::host_reports (
+  Optional[String] $proxy_name = undef,
+  Stdlib::Absolutepath $spool_dir = '/var/lib/foreman-proxy/host_reports',
+  Boolean $enabled = true,
+  Foreman_proxy::ListenOn $listen_on = 'https',
+  Optional[String] $version = undef,
+) {
+  $reported_proxy_hostname = pick($proxy_name, $foreman_proxy::registered_name)
+
+  file_line { "set report type in proxy ansible.cfg":
+    ensure => present,
+    path => "${foreman_proxy::config_dir}/ansible.cfg",
+    line => "report_type = proxy",
```

Comment:
If users install this plugin, I guess they need to be aware that it will work only with this setting in `ansible.cfg`, thus switching the method automatically.